src/prefect/filesystems.py

```diff
@@ -114,10 +114,6 @@ def _resolve_path(self, path: str) -> Path:
             resolved_path = basepath / resolved_path
         else:
             resolved_path = resolved_path.resolve()
-            if basepath not in resolved_path.parents and (basepath != resolved_path):
-                raise ValueError(
-                    f"Provided path {resolved_path} is outside of the base path {basepath}."
-                )
```

Comment:
this logic seemed unnecessary?

Reply:
I think it's to guard against cases where a `basepath` is configured and a method like `write_path` is given a path that tries to access outside the configured `basepath` (e.g. `../some/other/folder`)

Reply:
So this is subtle but this logic basically prevents absolute paths from being useful, since the loaded local storage reference needs to point to the same location as the stored absolute path

Reply:
I think you'd avoid this check if you didn't set a `basepath` when loading stored results. I don't think a `basepath` makes sense in that case since you already have an absolute path.

Reply:
Right but the code paths don't make that easy - the block is pre-loaded by the result factory and used to load results from their storage keys; so to keep this error i'd end up with a bunch of odd `try/except`s that explicitly work around the error, which felt counter productive to me?

Reply:
Ultimately this whole class is just a thin wrapper around a local filesystem, I'm not convinced it needs any business logic or guards

Reply:
OK so I kept this logic as opt-in validate logic for _writing_ but not for reading.